src/csharp/Microsoft.Spark/Sql/GenericRow.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.Sql.Types;
+
+namespace Microsoft.Spark.Sql
+{
+    /// <summary>
+    /// Represents a row object in RDD, equivalent to GenericRow in Spark.
+    /// </summary>
+    public sealed class GenericRow
+    {
+        /// <summary>
+        /// Constructor for the Row class.
+        /// </summary>
+        /// <param name="values">Column values for a row</param>        
+        internal GenericRow(object[] values)
+        {
+            Values = values;   
+            //TODO:
+            //Convert() -> implement type checking for not implemented exception
+        }
+       
+        /// <summary>
+        /// Values representing this row.
+        /// </summary>
+        public object[] Values { get; }
+
+        /// <summary>
+        /// Returns the number of columns in this row.
+        /// </summary>
+        /// <returns>Number of columns in this row</returns>
+        public int Size() => Values.Length;
+
+        /// <summary>
+        /// Returns the column value at the given index.
+        /// </summary>
+        /// <param name="index">Index to look up</param>
+        /// <returns>A column value</returns>
+        public object this[int index] => Get(index);
+
+        /// <summary>
+        /// Returns the column value at the given index.
+        /// </summary>
+        /// <param name="index">Index to look up</param>
+        /// <returns>A column value</returns>
+        public object Get(int index)
+        {
+            if (index >= Size())
+            {
+                throw new IndexOutOfRangeException($"index ({index}) >= column counts ({Size()})");
+            }
+            else if (index < 0)
+            {
+                throw new IndexOutOfRangeException($"index ({index}) < 0)");
+            }
+
+            return Values[index];
+        }
+
+        ///// <summary>
+        ///// Returns the column value whose column name is given.
+        ///// </summary>
+        ///// <param name="columnName">Column name to look up</param>
+        ///// <returns>A column value</returns>
+        //public object Get(string columnName) =>
```

Comment:
Any reason to comment Get function out?

Reply:
Wanted to check this with @imback82 , @suhsteve , as this takes in a column name as argument and GenericRow has no schema. I think we don't need this function but wanted to verify.